ui/app/src/components/DatasourceList/DatasourceEditorForm.tsx

```diff
@@ -0,0 +1,245 @@
+            {saveAction}
+          </Button>
+          <Button color="secondary" variant="outlined" onClick={handleCancel}>
+            Cancel
+          </Button>
+          {onDelete && (
+            <Button disabled={isReadonly} color="error" variant="outlined" onClick={onDelete}>
+              Delete
+            </Button>
+          )}
+        </Stack>
+      </Box>
+      <Box padding={2} sx={{ overflowY: 'scroll' }}>
+        <Grid container spacing={2} mb={2}>
+          <Grid item xs={4}>
+            <TextField
+              required
+              error={!!validation.name}
+              fullWidth
+              label="Name"
+              value={state.metadata.name}
+              InputProps={{
+                readOnly: isReadonly,
+              }}
+              helperText={validation.name}
+              onChange={(v) => {
+                setState((draft) => {
+                  draft.metadata.name = v.target.value;
+                });
+              }}
+            />
+          </Grid>
+          <Grid item xs={8}>
+            <TextField
+              fullWidth
+              label="Display Label"
+              value={state.spec.display?.name}
+              InputProps={{
+                readOnly: isReadonly,
+              }}
+              onChange={(v) => {
+                setState((draft) => {
+                  if (draft.spec.display) {
+                    draft.spec.display.name = v.target.value;
+                  }
+                });
+              }}
+            />
+          </Grid>
+          <Grid item xs={12}>
+            <TextField
+              fullWidth
+              label="Description"
+              value={state.spec.display?.description}
+              InputProps={{
+                readOnly: isReadonly,
+              }}
+              onChange={(v) => {
+                setState((draft) => {
+                  if (draft.spec.display) {
+                    draft.spec.display.description = v.target.value;
+                  }
+                });
+              }}
+            />
+          </Grid>
+          <Grid item xs={6}>
+            <FormControl fullWidth>
+              {/* TODO: How to ensure ids are unique? */}
+              <InputLabel id="datasource-default-label">Default</InputLabel>
+              <Select
+                labelId="datasource-default-label"
+                id="datasource-default-select"
+                label="Default"
+                value={state.spec.default == true ? 'yes' : 'no'}
+                readOnly={isReadonly}
+                onChange={(v) => {
+                  setState((draft) => {
+                    draft.spec.default = v.target.value === 'yes';
+                  });
+                }}
+                sx={{ width: '100%' }}
+              >
+                <MenuItem value="yes">Yes</MenuItem>
+                <MenuItem value="no">No</MenuItem>
+              </Select>
+            </FormControl>
+            <FormHelperText>
```

Comment:
nit: Help text here gets tight with control above:
<img width="483" alt="image" src="https://github.com/perses/perses/assets/9369625/8212e5ec-7683-4803-be7f-1f4f178cca4f">

Reply:
This form control was changed to a switch, the spacing is better now :)

![image](https://github.com/perses/perses/assets/7058693/ffcf2fd8-0b91-4973-b5ee-635b2de4e986)